From khali@linux-fr.org Mon Sep 26 02:14:46 2005
Date: Sun, 25 Sep 2005 16:56:43 +0200
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: Hideki Iwamoto <h-iwamoto@kit.hi-ho.ne.jp>
Subject: i2c: Fix union i2c_smbus_data definition
Message-Id: <20050925165643.151d5da1.khali@linux-fr.org>

From: Hideki Iwamoto <h-iwamoto@kit.hi-ho.ne.jp>

The i2c_smbus_data union block member has a comment stating that an
extra byte is required for SMBus Block Process Call transactions. This
has been true for three weeks around June 2002, but no more since, so
it is about time that we drop this comment and fix the definition.

From: Hideki Iwamoto <h-iwamoto@kit.hi-ho.ne.jp>
Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


 include/linux/i2c.h |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

---
--- gregkh-2.6.orig/include/linux/i2c.h
+++ gregkh-2.6/include/linux/i2c.h
@@ -454,8 +454,7 @@ struct i2c_msg {
 union i2c_smbus_data {
 	__u8 byte;
 	__u16 word;
-	__u8 block[I2C_SMBUS_BLOCK_MAX + 3]; /* block[0] is used for length */
-                          /* one more for read length in block process call */
+	__u8 block[I2C_SMBUS_BLOCK_MAX + 2]; /* block[0] is used for length */
 	                                            /* and one more for PEC */
 };
 
